test/e2e/performanceprofile/functests/7_performance_kubelet_node/cgroups.go

```diff
@@ -76,6 +214,325 @@ var _ = Describe("[performance] Cgroups and affinity", Ordered, func() {
+				checkCgroupCmd := []string{"cat", cgroupPath}
+				cgroupInfo, err := nodes.ExecCommandOnNode(checkCgroupCmd, workerRTNode)
+				ok, err := regexp.MatchString(processCgroups, cgroupInfo)
+				if err != nil {
+					testlog.Error(err.Error())
+				}
+				if ok {
+					testlog.Info(processCgroups)
+				}
+			}
+		})
+	})
+
+	Describe("Affinity", func() {
+		Context("ovn-kubenode Pods affinity ", func() {
+			BeforeEach(func() {
+				initialProfile = profile.DeepCopy()
+			})
+			It("[test_id:64100] matches with ovs process affinity", func() {
+				ovnPod, err := getOvnPod(workerRTNode)
+				Expect(err).ToNot(HaveOccurred(), "Unable to get ovnPod")
+				ovnContainersids, err := getOvnPodContainers(&ovnPod)
+				Expect(err).ToNot(HaveOccurred())
+				// Generally there are many containers inside a kubenode pods
+				// we don't need to check cpus used by all the containers
+				cpus := getCpusUsedByOvnContainer(ovnContainersids[0])
+				testlog.Infof("Cpus used by ovn Containers are %s", cpus)
+				pidList, err := getOVSServicesPid(workerRTNode)
+				Expect(err).ToNot(HaveOccurred())
+
+				for _, pid := range pidList {
+					cmd := []string{"/bin/bash", "-c", fmt.Sprintf("grep Cpus_allowed_list /proc/%s/status | awk '{print $2}'", pid)}
+					cpusOfovsServices, err := nodes.ExecCommandOnNode(cmd, workerRTNode)
+					Expect(cpusOfovsServices).ToNot(BeEmpty(), "unable to fetch cpus of ovs pid %s", pid)
+					testlog.Infof("cpus used by ovs services are: %s", cpusOfovsServices)
+					Expect(cpus == cpusOfovsServices).To(BeTrue(), "affinity of ovn kube node pods(%s) do not match with ovservices(%s)", cpus, cpusOfovsServices)
+					if err != nil {
+						testlog.Error(err.Error())
+					}
+				}
+			})
+
+			It("[test_id:64101] Creating gu pods modifies affinity of ovs", func() {
+				var testpod *corev1.Pod
+				var err error
+				testpod = pods.GetTestPod()
+				testpod.Namespace = testutils.NamespaceTesting
+				testpod.Spec.Containers[0].Resources = corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						corev1.ResourceCPU:    resource.MustParse("2"),
+						corev1.ResourceMemory: resource.MustParse("200Mi"),
+					},
+				}
+
+				testpod.Spec.NodeSelector = map[string]string{testutils.LabelHostname: workerRTNode.Name}
+				err = testclient.Client.Create(context.TODO(), testpod)
+				Expect(err).ToNot(HaveOccurred())
+				testpod, err = pods.WaitForCondition(client.ObjectKeyFromObject(testpod), corev1.PodConditionType(corev1.PodReady), corev1.ConditionTrue, 5*time.Minute)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(testpod.Status.QOSClass).To(Equal(corev1.PodQOSGuaranteed))
+				By("Getting the container cpuset.cpus cgroup")
+				containerID, err := pods.GetContainerIDByName(testpod, "test")
+				Expect(err).ToNot(HaveOccurred())
+
+				containerCgroup := ""
+				Eventually(func() string {
+					cmd := []string{"/bin/bash", "-c", fmt.Sprintf("find /rootfs/sys/fs/cgroup/cpuset/ -name *%s*", containerID)}
```

Comment:
As 4.15 will be using cgroupv2, I believe we need to have some generic function to switch respectively according to what cgroup the system is configured to, yet I believe this is part of next release.

Reply:
+1